### PR TITLE
[RDY] Support more audio formats for background music

### LINKED ISSUE #540: No MP3 Audio on OSX 10.8.3 or 64bit windows
_From [darkome...@gmail.com](https://code.google.com/u/111195278708016000746/) on April 04, 2013 17:00:13_

What steps will reproduce the problem? 1. Uncomment config file audio_mp3 = [[/Applications/CorsixTH/Music]]
2. Unzip MP3 from ThemeHospital_ZRRemix.zip in the Music folder
3. No any music when I launche CorsixTH 

Terminal output : 
CorsixTH.app/Contents/Resources/Lua/audio.lua:436: Could not load music file '/Applications/CorsixTH/Music/Theme Hospital - Mend (ZR Remix).mp3' (Unrecognized music format)
stack traceback:
    [C]: in function 'yield'
    CorsixTH.app/Contents/Resources/Lua/app.lua:690: in function <CorsixTH.app/Contents/Resources/Lua/app.lua:685>

What version of CorsixTH are you using (e.g. "Beta 2", "SVN r123 ")? 
0.20 What operating system / compile settings are you using? OSX 10.8.3, precompiled 64 and 32bits version

_Original issue: http://code.google.com/p/corsix-th/issues/detail?id=1569_

### PR DESCRIPTION
**Pull request by tobylane**
This plays major file types from [this sdl2_mixer page](https://www.libsdl.org/projects/SDL_mixer/docs/SDL_mixer_2.html), others could be added. I'm not sure on the names waveform and instructional.

The support that the second commit talks about is broad, for me it's `brew install libvorbis; brew reinstall sdl2_mixer --with-mpg123 --with-flac --with-libmikmod --with-fluid-synth`. The equivalent could be done for the deps repo.

Fixes #540 and #821.

### CODE REVIEW

File: CorsixTH/Lua/config_finder.lua [hunk 1]
```diff
@@ -413,10 +413,8 @@ new_graphics_folder = nil -- [[X:\ThemeHospital\Graphics]]
...
 -------------------------------------------------------------------------------------------------------------------------
--- High quality (MP3 rather than MIDI) audio replacements.
--- If you want to listen to high quality MP3 audio rather than the original XMI
--- (MIDI) audio, then follow these steps:
---  1) Find MP3 versions of the original tracks (for example the remixes by ZR
+-- If you want to listen to non-Theme-Hospital music, then follow these steps:
+--  1) Find updated versions of the original tracks (for example the remixes by ZR
 --     from http://www.a-base.dds.nl/temp/ThemeHospital_ZRRemix.zip ) or any
 --     other music you want to listen to.
 --  2) Ensure that SMPEG.dll (or equivalent for your platform) is present.
```
Comment (TheCycoONE) [inline CorsixTH/Lua/config_finder.lua]: `SMPEG.dll` is not used on any platform anymore, and different libraries are needed for different sound formats. This comment should be generalized.

File: WindowsInstaller/config_template.txt [hunk 2]
```diff
@@ -130,9 +130,7 @@ audio_buffer_size = 2048
 -------------------------------------------------------------------------------
 
 -------------------------------------------------------------------------------
--- High quality (MP3 rather than MIDI) audio replacements.
--- If you want to listen to high quality MP3 audio rather than the original XMI
--- (MIDI) audio, then follow these steps:
+-- If you want to listen to non-Theme-Hospital music, then follow these steps:
 --  1) Find MP3 versions of the original tracks (for example the remixes by ZR
```
Comment (TheCycoONE) [inline WindowsInstaller/config_template.txt]: MP3 comment here needs to be generalized for more music formats

File: CorsixTH/Lua/config_finder.lua (hunk 1 above)
Comment (TheCycoONE) [reply CorsixTH/Lua/config_finder.lua]: Actually remove that line completely. This is for end users, and we will ship with all relevant libraries.

**Comment by tobylane** [comment]
The third comment removes a persistable function. I haven't done this before, is there something extra in removing them, or is that only for functions in a savegame?

**Comment by TheCycoONE** [comment]
Normally you can't remove persistable functions, but that one is outside of any possible save, so I'm not sure why it was marked that way in the first place.

If your save games load, it should be good.

**Comment by TheCycoONE** [comment]
What's the state of this @tobylane

**Comment by tobylane** [comment]
I've removed the other commit and rebased. Ready now.

**Comment by TheCycoONE** [comment]
Tested 👍

**Pull request merged**